extensions/json-schema/rest/src/main/java/org/apache/unomi/schema/rest/JsonSchemaEndPoint.java

```diff
@@ -114,4 +116,22 @@ public Response save(String jsonSchema) {
     public boolean remove(String id) {
         return schemaService.deleteSchema(id);
     }
+
+    /**
+     * Being able to validate a given event is useful when you want to develop custom events and associated schemas
+     * @param event the event to be validated
+     * @return Validation error messages if there is some
+     */
+    @POST
+    @Produces(MediaType.APPLICATION_JSON + ";charset=UTF-8")
+    @Consumes({ MediaType.TEXT_PLAIN, MediaType.APPLICATION_JSON })
+    @Path("/validateEvent")
+    public Collection<ValidationError> validateEvent(String event) {
+        try {
+            return schemaService.validateEvent(event);
```

Comment:
There is small inconsistency in this method, we should return an error code always when the schema is invalid.
Ideally, all the errors should be handled by the Exception (if possible and have `schemaService.validateEvent()` to return a boolean. The `Collection<ValidationError>` can be carried by the ValidateException. 
If we don't want to do such change, at least we should return an error code in case of invalid event.

Reply:
I'm not sure, the goal of this endpoint is to return the validation errors.
So even if there is some, the request is valid and should be a 200.
The response code will be impacted if something goes wrong before doing the validation or in case of unexpected error during the validation process, that's why there is a difference between the exception and the validation errors.
The separation have been done on purpose.

ValidationException: Something unexpected happen and the validation was not able to be performed
ValidationError: The validation process is done successfully and we have informations on why the event does not respect the schemas.